components/script/dom/document.rs

```diff
@@ -1634,7 +1636,35 @@ impl Document {
         ).unwrap();
 
         // Step 8.
-        // TODO: pageshow event.
+        let document = Trusted::new(self);
+        if !self.page_showing.get() && document.root().browsing_context().is_some() {
```

Comment:
These check should be part of the queued steps instead.

Reply:
Only the page_showing one I think, but does that make a difference since it would be an early return in the task?

Reply:
If the document state changes in such a way that the task shouldn't execute later, it could make a difference.